src/expr.cpp

```diff
@@ -2647,6 +2648,66 @@ void BinaryExpr::Print() const {
     pos.Print();
 }
 
+std::pair<llvm::Constant *, bool> BinaryExpr::GetConstant(const Type *type) const {
+
+    // Are we doing something like (basePtr + offset)[...] = ... for a Global
+    // Variable
+    if (!GetLValueType())
+        return std::pair<llvm::Constant *, bool>(NULL, false);
+
+    // We are limiting cases to just addition and subtraction involving
+    // pointer addresses
+    // Case 1 : first argument is a pointer address.
+    // In this case as long as the second argument is a constant value, we are fine
+    // Case 2 : second argument is a pointer address.
+    // In this case, it has to be an addition with first argument as
+    // a constant value.
+    if (!((op == BinaryExpr::Op::Add) || (op == BinaryExpr::Op::Sub)))
+        return std::pair<llvm::Constant *, bool>(NULL, false);
+    if (op == BinaryExpr::Op::Sub) {
+        // Ignore cases where subtrahend is a PointerType
+        // Eg. b - 5 is valid but 5 - b is not.
+        if (CastType<PointerType>(arg1->GetType()))
+            return std::pair<llvm::Constant *, bool>(NULL, false);
+    }
+
+    // 'isNotValidForMultiTargetGlobal' is required to let the caller know
+    // that the llvm::constant value returned cannot be used in case of
+    // multi-target compilation.
```

Comment:
add "for initialization of globals. This is due to different constant values for different targets, i.e. computation involving sizeof() of varying types"

Reply:
Done